File: AMS_资讯信息组件/code/client/source/YssSecInformation.Support/Sv/Pojo/FiPay.cs
﻿using FAST.Core.Util;
using FAST.Common.Service.Pojo;
using FAST.Common.Service.Pojo.Base;
using FAST.Core.Exceptions;


using FAST.Core.Context;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Newtonsoft.Json;

namespace YssSecInformation.Support.Sv.Pojo
{
    /// <summary>
    /// 债券历史付息pojo类
    /// </summary>
    public class FiPay : EnclosurePojo
    {
        /// <summary>
        /// 债券代码
        /// </summary>
        private string c_SEC_CODE = "";

        /// <summary>
        /// 调息日期
        /// </summary>
        private string d_ADJ = "";

        /// <summary>
        ///  票面利率
        /// </summary>
        private decimal n_COUP_RATE = 0;

        /// <summary>
        ///  剩余本金
        /// </summary>
        private decimal n_REPAY_AMOUNT = 0;

        /// <summary>
        ///  偿还数量
        /// </summary>
        private decimal n_REM_COR = 0;

        /// <summary>
        ///  本次起息日
        /// </summary>
        private string d_BEGIN = "";

        /// <summary>
        /// 本次截息日
        /// </summary>
        private string d_END = "";

        /// <summary>
        ///  描述
        /// </summary>
        private string c_DESC = "";

        /// <summary>
        /// 周期新利率 
        /// </summary>
        private string c_DV_BOOL_TYPE = "";

        /// <summary>
        /// 交易市场
        /// </summary>
        private string c_MKT_CODE = "";

        
        /// <summary>
        /// 数据来源
        ///edit tdf BUG #115588 [紧急]浮动利率债券没有清算出财汇资讯数据 20150715
        /// </summary>
        private string c_DATA_IDF = "";

        /// <summary>
        /// 是否期内还本
        /// </summary>
        private decimal n_QNHB = 0;

        /// <summary>
        /// 期内还本日期
        /// </summary>
        private string d_QNHB = "";


        /// <summary>
        /// 属性: 债券代码 
        /// </summary>
        [JsonProperty(PropertyName = "c_SEC_CODE")]
        public string C_SEC_CODE
        {
            set { c_SEC_CODE = value; }

            get { return c_SEC_CODE; }
        }

        /// <summary>
        /// 属性: 调息日期 
        /// </summary>
        [JsonProperty(PropertyName = "d_ADJ")]
        public string D_ADJ
        {
            set { d_ADJ = value; }

            get { return d_ADJ; }
        }

        /// <summary>
        /// 属性: 票面利率 
        /// </summary>
        [JsonProperty(PropertyName = "n_COUP_RATE")]
        public decimal N_COUP_RATE
        {
            set { n_COUP_RATE = value; }

            get { return n_COUP_RATE; }
        }

        /// <summary>
        /// 属性: 剩余本金 
        /// </summary>
        [JsonProperty(PropertyName = "n_REM_COR")]
        public decimal N_REM_COR
        {
            set { n_REM_COR = value; }

            get { return n_REM_COR; }
        }

        /// <summary>
        /// 属性: 偿还数量
        /// </summary>
        [JsonProperty(PropertyName = "n_REPAY_AMOUNT")]
        public decimal N_REPAY_AMOUNT
        {
            set { n_REPAY_AMOUNT = value; }

            get { return n_REPAY_AMOUNT; }
        }

        /// <summary>
        /// 属性: 本次起息日 
        /// </summary>
        [JsonProperty(PropertyName = "d_BEGIN")]
        public string D_BEGIN
        {
            set { d_BEGIN = value; }

            get { return d_BEGIN; }
        }

        /// <summary>
        /// 属性: 本次截息日 
        /// </summary>
        [JsonProperty(PropertyName = "d_END")]
        public string D_END
        {
            set { d_END = value; }

            get { return d_END; }
        }

        /// <summary>
        /// 属性: 描述 
        /// </summary>
        [JsonProperty(PropertyName = "c_DESC")]
        public string C_DESC
        {
            set { c_DESC = value; }

            get { return c_DESC; }
        }

        /// <summary>
        /// 属性: 周期新利率 
        /// </summary>
        [JsonProperty(PropertyName = "c_DV_BOOL_TYPE")]
        public string C_DV_BOOL_TYPE
        {
            set { c_DV_BOOL_TYPE = value; }

            get { return c_DV_BOOL_TYPE; }
        }

        /// <summary>
        /// 属性: 交易市场 
        /// </summary>
        [JsonProperty(PropertyName = "c_MKT_CODE")]
        public string C_MKT_CODE
        {
            set { c_MKT_CODE = value; }

            get { return c_MKT_CODE; }
        }

        /// <summary>
        /// 属性：数据来源
        /// edit tdf BUG #115588 [紧急]浮动利率债券没有清算出财汇资讯数据 20150715
        /// </summary>
        [JsonProperty(PropertyName = "c_DATA_IDF")]
        public string C_DATA_IDF
        {
            get { return c_DATA_IDF; }
            set { c_DATA_IDF = value; }
        }

        /// <summary>
        /// 属性：是否期内还本
        /// </summary>
        [JsonProperty(PropertyName = "n_QNHB")]
        public decimal N_QNHB
        {
            get { return n_QNHB; }
            set { n_QNHB = value; }
        }

        /// <summary>
        /// 属性：期内还本日期
        /// </summary>
        [JsonProperty(PropertyName = "d_QNHB")]
        public string D_QNHB
        {
            get { return d_QNHB; }
            set { d_QNHB = value; }
        }
    }
}

File: AMS_资讯信息组件/code/server/source/com.yss.ams.sec.information.support/src/main/java/com/yss/ams/sec/information/support/modules/sv/fipay/pojo/FiPay.java
package com.yss.ams.sec.information.support.modules.sv.fipay.pojo;


import java.math.BigDecimal;

import com.yss.framework.api.common.co.AuditableParamPojo;

/**
 * 债券付息信息实体类，解析和拼接数据
 * 
 * @author yh 2010.12.31
 * @version V4.5.0.1 债券付息信息实体类，实现具体的解析和拼接数据 lyh 2011.1.6 V4.5.0.2 ---修改记录---
 *          修改人：wuwenlan 20110223 当前版本：v4.5.03 修改描述： 类的属性命名修改
 *          修改解析方法，拼接方法和获取数据库字段值
 * 
 *          4.5.0.4 修改日期 2011-03-11 修改简介：添加//周期新利率 字段
 * @author chenyoulong
 */

/**
 * 债券付息信息pojo类
 * @author yuankai
 * 资讯信息拆分	2017.7.3 STORY #42948 资讯信息管理组件化拆分
 *
 */
public class FiPay extends AuditableParamPojo {

	/**
	 * 
	 */
	private static final long serialVersionUID = 3167945825423304062L;
	/**
	* 债券代码 
	*/
	  private String c_SEC_CODE = "";

	/**
	* 调息日期 
	*/
	  private String d_ADJ = "";

	/**
	* 票面利率 
	*/
	  private BigDecimal n_COUP_RATE = BigDecimal.ZERO;

	/**
	* 剩余本金 
	*/
	  private BigDecimal n_REM_COR = BigDecimal.ZERO;
	  
	/**
	* 偿还数量
	*/
	  private BigDecimal n_REPAY_AMOUNT = BigDecimal.ZERO;

	/**
	* 本次起息日 
	*/
	  private String d_BEGIN = "";

	/**
	* 本次截息日 
	*/
	  private String d_END = "";

	/**
	* 描述 
	*/
	  private String c_DESC = "";

	/**
	* 周期新利率 
	*/
	  private String c_DV_BOOL_TYPE = "";
	  
   /**
    * 申购期间收益
    */
	  private String c_SGQJSY = "";

	/**
	* 交易市场 
	*/
	  private String c_MKT_CODE = "";
	//  edit tdf BUG #115588 [紧急]浮动利率债券没有清算出财汇资讯数据 20150715
	 /**
	  * 数据来源
	 */
	  private String c_DATA_IDF = "";
	  
	/**
	 * 是否期内还本
	 */
	private int n_QNHB = 0 ;
	  
	/**
	 * 期内还本日期
	 */
	private String d_QNHB = "";
	  
	public String getC_SEC_CODE() {
		return c_SEC_CODE;
	}

	public void setC_SEC_CODE(String cSECCODE) {
		c_SEC_CODE = cSECCODE;
	}

	public String getD_ADJ() {
		return d_ADJ;
	}

	public void setD_ADJ(String dADJ) {
		d_ADJ = dADJ;
	}

	public BigDecimal getN_COUP_RATE() {
		return n_COUP_RATE;
	}

	public void setN_COUP_RATE(BigDecimal nCOUPRATE) {
		n_COUP_RATE = nCOUPRATE;
	}

	public BigDecimal getN_REM_COR() {
		return n_REM_COR;
	}

	public void setN_REM_COR(BigDecimal nREMCOR) {
		n_REM_COR = nREMCOR;
	}

	public String getD_BEGIN() {
		return d_BEGIN;
	}

	public void setD_BEGIN(String dBEGIN) {
		d_BEGIN = dBEGIN;
	}

	public String getD_END() {
		return d_END;
	}

	public void setD_END(String dEND) {
		d_END = dEND;
	}

	public String getC_DESC() {
		return c_DESC;
	}

	public void setC_DESC(String cDESC) {
		c_DESC = cDESC;
	}

	public String getC_DV_BOOL_TYPE() {
		return c_DV_BOOL_TYPE;
	}

	public void setC_DV_BOOL_TYPE(String cDVBOOLTYPE) {
		c_DV_BOOL_TYPE = cDVBOOLTYPE;
	}

	public String getC_MKT_CODE() {
		return c_MKT_CODE;
	}

	public void setC_MKT_CODE(String cMKTCODE) {
		c_MKT_CODE = cMKTCODE;
	}

	public static long getSerialversionuid() {
		return serialVersionUID;
	}
	
	public BigDecimal getN_REPAY_AMOUNT() {
		return n_REPAY_AMOUNT;
	}

	public void setN_REPAY_AMOUNT(BigDecimal n_REPAY_AMOUNT) {
		this.n_REPAY_AMOUNT = n_REPAY_AMOUNT;
	}

//  edit tdf BUG #115588 [紧急]浮动利率债券没有清算出财汇资讯数据 20150715
	public String getC_DATA_IDF() {
		return c_DATA_IDF;
	}

	public void setC_DATA_IDF(String c_DATA_IDF) {
		this.c_DATA_IDF = c_DATA_IDF;
	}

	public String getC_SGQJSY() {
		return c_SGQJSY;
	}

	public void setC_SGQJSY(String cSGQJSY) {
		this.c_SGQJSY = cSGQJSY;
	}

	public int getN_QNHB() {
		return n_QNHB;
	}

	public void setN_QNHB(int n_QNHB) {
		this.n_QNHB = n_QNHB;
	}

	public String getD_QNHB() {
		return d_QNHB;
	}

	public void setD_QNHB(String d_QNHB) {
		this.d_QNHB = d_QNHB;
	}



}
